libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,511 @@
+                                                 sizeof( SocketsSockaddr_t ) );
+
+        if( transportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to establish new connection. Socket status: %d.", transportSocketStatus ) );
+            returnStatus = TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE;
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        /* Configure send and receive timeouts for the socket. */
+        transportSocketStatus = transportTimeoutSetup( tcpSocket, pSocketsConfig->sendTimeoutMs, pSocketsConfig->recvTimeoutMs );
+
+        if( transportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to configure send and receive timeouts for socket: Status=%d.", transportSocketStatus ) );
+            returnStatus = TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR;
+        }
+    }
+
+    if( returnStatus == TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        /* Set the socket in the network context. */
+        pNetworkContext->pContext = tcpSocket;
+    }
+    else
+    {
+        /* Clean up socket on failure. */
+        ( void ) SOCKETS_Close( tcpSocket );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+TransportSocketStatus_t SecureSocketsTransport_Connect( NetworkContext_t * pNetworkContext,
+                                                        const ServerInfo_t * pServerInfo,
+                                                        const SocketsConfig_t * pSocketsConfig )
+{
+    TransportSocketStatus_t returnStatus = TRANSPORT_SOCKET_STATUS_SUCCESS;
+
+    /* Sanity checks for input parameters. */
+    if( pSocketsConfig == NULL )
+    {
+        LogError( ( "Parameter check failed: pSocketsConfig is NULL." ) );
+        returnStatus = TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER;
+    }
+    else if( pServerInfo == NULL )
+    {
+        LogError( ( "Parameter check failed: pServerInfo is NULL." ) );
+        returnStatus = TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER;
+    }
+    else if( pServerInfo->pHostName == NULL )
+    {
+        LogError( ( "Parameter check failed: pServerInfo->pHostName is NULL." ) );
+        returnStatus = TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER;
+    }
+    else if( pServerInfo->hostNameLength == 0UL )
+    {
+        LogError( ( "Parameter check failed: hostNameLength must be greater than 0." ) );
+        returnStatus = TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER;
+    }
+    else
+    {
+        /* Establish the TCP connection. */
+        returnStatus = establishConnect( pNetworkContext,
+                                         pServerInfo,
+                                         pSocketsConfig );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+int32_t SecureSocketsTransport_Disconnect( const NetworkContext_t * pNetworkContext )
```

Comment:
Should this return value be TransportSocketStatus_t?

Reply:
fixed in c5a6dbb